aptos_sdk/async_client.py

```diff
@@ -568,24 +568,31 @@ async def transaction_pending(self, txn_hash: str) -> bool:
             raise ApiError(response.text, response.status_code)
         return response.json()["type"] == "pending_transaction"
 
+    async def _wait_for_txn(self, txn_hash: str) -> None:
+        while True:
+            response = await self._get(endpoint=f"transactions/wait_by_hash/{txn_hash}")
+            if response.status_code >= 400:
+                raise ApiError(response.text, response.status_code)
+            is_pending = response.json()["type"] == "pending_transaction"
+            if not is_pending:
+                assert (
+                    "success" in response.json() and response.json()["success"]
+                ), f"{response.text} - {txn_hash}"
+                return
+
```

Comment:
The nerd in me can't help but want to optimize this...
```suggestion
        is_pending = True
        while is_pending:
            response = await self._get(endpoint=f"transactions/wait_by_hash/{txn_hash}")
            if response.status_code >= 400:
                raise ApiError(response.text, response.status_code)
            is_pending = response.json()["type"] == "pending_transaction"
        assert (
            "success" in response.json() and response.json()["success"]
        ), f"{response.text} - {txn_hash}"

```

alternatively, we could 
```
if not is_pending:
    break
```